datafusion/spark/src/function/hash/crc32.rs

```diff
@@ -124,11 +113,12 @@ fn spark_crc32(args: &[ArrayRef]) -> Result<ArrayRef> {
             let input = as_binary_view_array(input)?;
             Ok(spark_crc32_impl(input.iter()))
         }
-        _ => {
-            exec_err!(
-                "Spark `crc32` function: argument must be binary or large binary, got {:?}",
-                input.data_type()
-            )
+        DataType::FixedSizeBinary(_) => {
+            let input = as_fixed_size_binary_array(input)?;
+            Ok(spark_crc32_impl(input.iter()))
+        }
+        dt => {
+            internal_err!("Unsupported data type for crc32: {dt}")
```

Comment:
I would suggest still keeping `exec_err`

Reply:
I switched them to `internal_err` as during normal execution these errors should never occur; the signature should guard against them for us, so only if something went wrong with the signature/coercion logic elsewhere in the engine would we run into these errors (hence considered an internal error)

Reply:
Right so idea of internal is highlight something really unexpected happening like a guard saying there is internal engine bug.

Will it trigger if user calls the crc32 with invalid param like 

```
select crc32(arrow_cast(null, 'Dictionary(Int32, Utf8)'))
```

Reply:
Added test case: https://github.com/apache/datafusion/pull/18662/commits/03539515ec3ded7c056477664b4e3a826b56f576

Raised arrow-rs issue: https://github.com/apache/arrow-rs/issues/8841